### PR TITLE
Merge block canvas changes to latest trunk

### LINKED ISSUE #16: Merge block canvas style updates into Course theme
[Some fairly substantial updates](https://github.com/Automattic/themes/pull/6640) were just made to the Block Canvas theme that we should bring over into the Course theme sooner rather than later.

### PR DESCRIPTION
**Pull request by LavaGolem**
**IMPORTANT NOTE**

This PR is a new PR on the same ticket as [this closed one](https://github.com/Automattic/sensei-theme/pull/22)

Fixes #16 

This PR is merging the latest changes from Block Canvas [PR ](https://github.com/Automattic/themes/pull/6640) 

This PR is created manually by comparing the changes and adding changes, making sure nothing that worked before is messed up. 

Note some changes are SKIPPED since we have implemented our own: 
- Updates spacing to match Figma definition
- Updates typography of headings

**Important note:** 
This change Updates min WP version to 6.0
Also, I've checked and for me, it was working only when I updated on 6.1

The new changes from Block Canvas

1. Underline on hover for the Header and Header navigation: 

https://user-images.githubusercontent.com/7208249/194382523-654af0f6-5b8a-4ca9-89cc-7a34ac0ba180.mov

2. Links within paragraphs, post tags and categories: underline on default, no underline on hover

https://user-images.githubusercontent.com/7208249/194384691-d2c25b1f-644d-4075-ae30-a9f86e80fe76.mov

3. Post-Author: underline by default and not on hover: 

https://user-images.githubusercontent.com/7208249/194492654-986ff299-5c39-4195-a2a9-4273587f9604.mov

4. Button - change the color on hover

https://user-images.githubusercontent.com/7208249/194494333-5fa9403d-6626-4c18-9ad3-58e36e085426.mov

5. Comment links - underline and no underline on hover

https://user-images.githubusercontent.com/7208249/194495431-bf172d0f-0f63-4795-b615-a2683ceb3875.mov

6. post date: no underline on default, underline on hover

https://user-images.githubusercontent.com/7208249/194496236-bcbdf8af-7e5d-4e20-aefb-12ea8a7e5d07.mov

7. Post title underline on hover

https://user-images.githubusercontent.com/7208249/194500329-c41acbf5-e472-4e73-bc6f-33a4f2a50785.mov

### CODE REVIEW

**Comment by LavaGolem** [comment]
@donnapep I've created a new PR since there were lots of new changes in the trunk and also the change between this and resolving all the conflicts was lots of work. 

I will address all the comments from the previous PR here as well and also left a comment there

File: patterns/footer.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Title: Default footer
+ * Slug: course/footer
```
Comment (LavaGolem) [inline patterns/footer.php:4]: @donnapep Removed block-canvas slug

File: patterns/footer.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+<!-- wp:group {"layout":{"inherit":true}} -->
+<div class="wp-block-group">
+    <!-- wp:group {"style":{"spacing":{"padding":{"top":"var(--wp--preset--spacing--60)","bottom":"var(--wp--preset--spacing--60)"}}}} -->
+    <div class="wp-block-group" style="padding-top:var(--wp--preset--spacing--60);padding-bottom:var(--wp--preset--spacing--60)">
+        <!-- wp:paragraph {"align":"center"} -->
+        <p class="has-text-align-center">
+            <?php
+            /* Translators: WordPress link. */
+            $wordpress_link = '<a href="' . esc_url( __( 'https://wordpress.org', 'course' ) ) . '" rel="nofollow">WordPress</a>';
+            echo sprintf(
```
Comment (LavaGolem) [inline patterns/footer.php:23]: Removed block-canvas metion

File: theme.json [hunk 3]
```diff
@@ -149,8 +149,8 @@
 		"blocks": {
 			"core/code": {
 				"border": {
-					"color": "#CCCCCC",
-					"radius": "0px",
+					"color": "var(--wp--preset--color--foreground)",
```
Comment (LavaGolem) [inline theme.json:168]: Added core/code change

File: theme.json [hunk 4]
```diff
@@ -222,17 +239,42 @@
...
+						":hover": {
+							"typography": {
+								"textDecoration": "underline"
+							}
+						}
+					}
 				}
 			},
 			"core/pullquote": {
 				"border": {
```
Comment (LavaGolem) [inline theme.json:299]: Pullquote change

File: theme.json [hunk 5]
```diff
@@ -297,6 +339,11 @@
 					"link": {
 						"typography": {
 							"textDecoration": "none"
+						},
```
Comment (LavaGolem) [inline theme.json:365]: site-title change

**Review by gabrielcaires — APPROVED** [review]
I fixed a small issue here [7d43686](https://github.com/Automattic/sensei-theme/pull/26/commits/7d43686cf8ad6cb274da07ebee430e6fe473400d)

Merging to avoid conflicts

**Pull request merged**